src/AWS.Logger.Core/Core/AWSLoggerCore.cs

```diff
@@ -79,16 +79,28 @@ public AWSLoggerCore(AWSLoggerConfig config, string logType)
             _config = config;
             _logType = logType;
 
-            var credentials = DetermineCredentials(config);
-
-            if (_config.Region != null)
-            {
-                _client = new AmazonCloudWatchLogsClient(credentials, Amazon.RegionEndpoint.GetBySystemName(_config.Region));
-            }
-            else
-            {
-                _client = new AmazonCloudWatchLogsClient(credentials);
-            }
+            if (!string.IsNullOrEmpty(_config.ServiceUrl))
```

Comment:
Optional alternative is to use this if and else clauses to construct the AmazonCloudWatchLogsConfig object. Then after the else block you can have just one section to construct the AmazonCloudWatchLogsClient with the AmazonCloudWatchLogsConfig. The region can be set on the AmazonCloudWatchLogsConfig object. Right now there is a bit of code duplication.

Reply:
Not sure I understand. When using the `ServiceUrl` then I create a custom `AmazonCloudWatchLogsConfig` instead of just letting `AmazonCloudWatchLogsClient` perform the automatic/default/best loading of config-object. I like the path where no custom `AmazonCloudWatchLogsConfig` is created.